newIDE/app/src/EventsFunctionsExtensionEditor/OptionsEditorDialog/UsersAutocomplete.js

```diff
@@ -0,0 +1,152 @@
+import { useState } from 'react';
+import { t } from '@lingui/macro';
+import SemiControlledMultiAutoComplete from '../../UI/SemiControlledMultiAutoComplete';
+import { type MessageDescriptor } from '../../Utils/i18n/MessageDescriptor.flow';
+import {
+  searchUserPublicProfilesByUsername,
+  type UserPublicProfile,
+  getUserPublicProfilesByIds,
+} from '../../Utils/GDevelopServices/User';
+
+import useForceUpdate from '../../Utils/UseForceUpdate';
+
+type Props = {|
+  userIds: gdVectorString,
+  floatingLabelText?: MessageDescriptor,
+  helperText: MessageDescriptor,
+|};
+
+type Option = {|
+  text: string,
+  value: string,
+|};
+
+const getErrorMessage = (error: ?Error) => {
+  if (error) return 'Error while loading users';
+};
+
+export const UsersAutocomplete = (props: Props) => {
+  const forceUpdate = useForceUpdate();
+  const [users, setUsers] = React.useState<Array<Option>>([]);
+  const [userInput, setUserInput] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [
+    completionUserPublicProfiles,
+    setCompletionUserPublicProfiles,
+  ] = React.useState<Array<UserPublicProfile>>([]);
+  const [error, setError] = React.useState(null);
+
+  // Recalculate if the userInput has changed.
+  const searchUserPublicProfiles = React.useCallback(
+    async () => {
+      setError(null);
+      if (!userInput) {
+        setCompletionUserPublicProfiles([]);
+        return;
+      }
+      try {
+        setLoading(true);
+        const userPublicProfiles = await searchUserPublicProfilesByUsername(
+          userInput
+        );
+        setCompletionUserPublicProfiles(userPublicProfiles);
+        setLoading(false);
+      } catch (err) {
+        setLoading(false);
+        setError(new Error('Error while loading users'));
+        console.log('Could not load the users: ', err);
+      }
+    },
+    [userInput]
+  );
+
+  // Call every time the userInput changes.
+  React.useEffect(
+    () => {
+      searchUserPublicProfiles();
+    },
+    [searchUserPublicProfiles, userInput]
+  );
+
+  // Do only once, the first time it is loaded.
+  const getUserPublicProfilesForAutocomplete = React.useCallback(
+    async () => {
+      setError(null);
+      const userIds = props.userIds.toJSArray();
+      if (!userIds.length) {
+        setUsers([]);
+        return;
+      }
+      try {
+        setLoading(true);
+        const userPublicProfilesByIds = await getUserPublicProfilesByIds(
+          userIds
+        );
+        setUsers(
+          Object.keys(userPublicProfilesByIds).map(userId => {
+            const userPublicProfile: UserPublicProfile =
+              userPublicProfilesByIds[userId];
+            return {
+              text: userPublicProfile.username || '(no username)',
+              value: userPublicProfile.id,
+            };
+          })
+        );
+        setLoading(false);
+      } catch (err) {
+        setLoading(false);
+        setError(new Error('Error while loading the users'));
```

Comment:
Same, store `err`

Reply:
(in an ideal world, errors would be strongly typed, so that we could have functions handling all kind of errors - at least be able to know what we're supposed to get (à la Java Exception) - unfortunately I think it's not really possible in Flow/TS? And this would need a strong integration between backends and frontends).